src/react/queries.ts

```diff
@@ -206,3 +222,60 @@ export const getAccountSeedsQuery = () => ({
   queryKey: ['AccountSeeds'],
   queryFn: getAccountSeeds,
 });
+
+export const getObjectsVersions = (
+  bucketName: string,
+  s3Client: S3,
+  enabled: boolean,
+  callback?: (listKeys: S3.ObjectIdentifierList) => void,
+  maxKeys?: number,
+): AWS_PAGINATED_QUERY<
+  ListObjectVersionsOutput,
+  S3.ObjectIdentifier,
+  AWSError,
+  ListObjectVersionsOutput
+> => ({
+  queryKey: ['objectVersions', bucketName],
+  queryFn: (_, marker) =>
+    s3Client
+      .listObjectVersions({
+        Bucket: bucketName,
+        KeyMarker: marker?.NextKeyMarker,
+        VersionIdMarker: marker?.NextVersionIdMarker,
+        MaxKeys: maxKeys,
+      })
+      .promise(),
+  enabled,
+  refetchOnWindowFocus: false,
+  refetchOnMount: false,
+  refetchOnReconnect: false,
+  onSuccess: (data) => {
+    const infiniteData = data as InfiniteData<ListObjectVersionsOutput>;
+    const pageParamsSize = infiniteData?.pageParams?.length;
+    const deleteObjects = [
+      ...(infiniteData?.pages?.[pageParamsSize - 1].DeleteMarkers || []).map(
+        (d) => ({
+          Key: d.Key || '',
+          VersionId: d.VersionId,
+        }),
+      ),
+      ...(infiniteData?.pages?.[pageParamsSize - 1].Versions || []).map(
+        (v) => ({
+          Key: v.Key || '',
+          VersionId: v.VersionId,
+        }),
+      ),
+    ];
+
+    if (deleteObjects?.length) {
+      callback?.(deleteObjects);
+    }
+  },
```

Comment:
nit: For future enhancement we might want to move this to a selector. Indeed it will allow to use the same cached data for object listing

Use query have the following props for this purpose:
```
select: (data: TData) => unknown
Optional
This option can be used to transform or select a part of the data returned by the query function. It affects the returned data value, but does not affect what gets stored in the query cache.


```

Reply:
we both use `select`  and `onSuccess` here because call a callback inside `select` will cause an infinit loop